dali/sasha/saqmon.cpp

```diff
@@ -222,36 +229,53 @@ class CSashaQMonitorServer: public ISashaServer, public Thread
         if (!initQueueNames(qmonprops->queryProp("@queues")))
             return 0;
         Owned<IRemoteConnection> conn = querySDS().connect("Status/Servers", myProcessSession(), 0, 100000);
-        if (!conn) {
+        if (!conn)
+        {
             OERRLOG("cannot connect to Status/Servers");
             return -1;
         }
         unsigned *qidlecount = new unsigned[qnames.ordinality()];
-        ForEachItemIn(i1,qnames) {
+        ForEachItemIn(i1,qnames)
+        {
             StringBuffer qname(qnames.item(i1));
             qname.append(".thor");
             queues.append(*createJobQueue(qname.str()));
             qidlecount[i1] = 0;
         }
         unsigned sleeptime = autoswitch?1:interval;
         unsigned moninter = interval;
-        while (!stopped) {
+        while (!stopped)
+        {
             stopsem.wait(60*1000*sleeptime);
             if (stopped)
                 break;
             moninter-=sleeptime;
-            if (autoswitch||(moninter==0)) { // always true at moment
-                try {
+            if (autoswitch||(moninter==0)) // always true at moment
+            {
+                try
+                {
                     conn->reload();
-                    ForEachItemIn(qi,qnames) {
+                    ForEachItemIn(qi,qnames)
+                    {
+                        StringBuffer thorQName;
                         const char *qname = qnames.item(qi);
-                        StringBuffer xpath;
-                        xpath.appendf("Server[@queue=\"%s.thor\"]/WorkUnit",qname);
-                        Owned<IPropertyTreeIterator> iter = conn->queryRoot()->getElements(xpath.str());
+                        getClusterThorQueueName(thorQName, qname);
+                        Owned<IPropertyTreeIterator> iter = conn->queryRoot()->getElements("Server[@queue]");
```

Comment:
@queue looks a bit strange.  Does that check that it has an entry?

Reply:
yes it's a qualifier without a result. Returns all "Server" nodes that have a @queue attribute.